modules/gapi/misc/python/test/test_gapi_infer_ov.py

```diff
@@ -0,0 +1,255 @@
+                                                                          .cfgInputTensorLayout("NHWC") \
+                                                                          .cfgResize(cv.INTER_LINEAR)
+
+            gapi_age, gapi_gender = comp.apply(cv.gin(batch_img), args=cv.gapi.compile_args(cv.gapi.networks(pp)))
+            gapi_gender = gapi_gender.squeeze()
+            gapi_age = gapi_age.squeeze()
+
+            # Check
+            self.assertEqual(0.0, cv.norm(ov_gender, gapi_gender, cv.NORM_INF))
+            self.assertEqual(0.0, cv.norm(ov_age, gapi_age, cv.NORM_INF))
+
+
+        def test_age_gender_infer_planar(self):
+            skip_if_openvino_not_available()
+
+            root_path  = '/omz_intel_models/intel/age-gender-recognition-retail-0013/FP32/age-gender-recognition-retail-0013'
+            model_path = self.find_file(root_path + '.xml',   [os.environ.get('OPENCV_DNN_TEST_DATA_PATH')])
+            bin_path   = self.find_file(root_path + '.bin',   [os.environ.get('OPENCV_DNN_TEST_DATA_PATH')])
+            device_id  = 'CPU'
+
+            img_path = self.find_file('cv/face/david2.jpg', [os.environ.get('OPENCV_TEST_DATA_PATH')])
+            img = cv.imread(img_path)
+            planar_img = np.transpose(img, (2, 0, 1))
+            planar_img = np.expand_dims(planar_img, 0)
+
+            # OpenVINO
+            core = Core()
+            model = core.read_model(model_path, bin_path)
+
+            ppp = PrePostProcessor(model)
+            ppp.input().tensor().set_element_type(Type.u8)                            \
+                                .set_spatial_static_shape(img.shape[0], img.shape[1]) \
+                                .set_layout(Layout("NCHW"))
+            ppp.input().preprocess().resize(ResizeAlgorithm.RESIZE_LINEAR)
+            model = ppp.build()
+
+            compiled_model = core.compile_model(model, device_id)
+            infer_request = compiled_model.create_infer_request()
+            results = infer_request.infer(planar_img)
+            ov_age = results['age_conv3'].squeeze()
+            ov_gender = results['prob'].squeeze()
+
+            # OpenCV G-API
+            g_in   = cv.GMat()
+            inputs = cv.GInferInputs()
+            inputs.setInput('data', g_in)
+
+            outputs  = cv.gapi.infer("net", inputs)
+            age_g    = outputs.at("age_conv3")
+            gender_g = outputs.at("prob")
+
+            comp = cv.GComputation(cv.GIn(g_in), cv.GOut(age_g, gender_g))
+            pp = cv.gapi.ov.params("net", model_path, bin_path, device_id).cfgResize(cv.INTER_LINEAR)
+
+            gapi_age, gapi_gender = comp.apply(cv.gin(planar_img), args=cv.gapi.compile_args(cv.gapi.networks(pp)))
+            gapi_gender = gapi_gender.squeeze()
+            gapi_age = gapi_age.squeeze()
+
+            # Check
+            self.assertEqual(0.0, cv.norm(ov_gender, gapi_gender, cv.NORM_INF))
+            self.assertEqual(0.0, cv.norm(ov_age, gapi_age, cv.NORM_INF))
+
```

Comment:
What about testing invalid cases? Where we get G-API or OV throwing an exception

Reply:
Well, there is not much `invalid` cases. The validation is only happens for `Image` case where the layout is being validated to be `NHWC` for other case there should be check that `mat` dims matches to `tensor` dims I added `TODO` for that